Dockerfile

```diff
@@ -5,4 +5,6 @@ RUN pip install -r /requirements.txt
 
 WORKDIR /opt/dagster/app
 
-COPY . /opt/dagster/app
\ No newline at end of file
+COPY . /opt/dagster/app
+
+RUN pip install -e .
```

Comment:
I think we normally recommend a different structure so that the pip installs are cached ahead of the code.

eg for hooli:

https://github.com/dagster-io/dagster-cloud-action/blob/main/src/Dockerfile.template

I think it would be helpful if, for this template, we had a bit more commentary on the Dockerfile. Untested, but I think something like this would make sense. We should also remove requirements.txt from this template if we aren't using it.


```
FROM python:3.8-slim

# Add any steps to install project system dependencies like java

# Add steps to install the Python dependencies for your Dagster project
# into the default Python on PATH
# For example, this project uses setup.py and we install the dependencies before
# copying the code to optimize the Docker cache
COPY  *setup.py
RUN pip install .

# Similar steps could be added for additional project dependencies like dbt deps

# Now we copy in the rest of the application code
WORKDIR /opt/dagster/app
COPY . /opt/dagster/app

# Make sure dagster-cloud is installed. Fail early here if not.
RUN dagster-cloud --version

# Thats it! You do not need an entrypoint command
```